src/libraries/Microsoft.Extensions.Options/ref/Microsoft.Extensions.Options.cs

```diff
@@ -297,6 +297,16 @@ public partial class ValidateOptionsResult
         public static Microsoft.Extensions.Options.ValidateOptionsResult Fail(System.Collections.Generic.IEnumerable<string> failures) { throw null; }
         public static Microsoft.Extensions.Options.ValidateOptionsResult Fail(string failureMessage) { throw null; }
     }
+    public class ValidateOptionsResultBuilder
+    {
+        public ValidateOptionsResultBuilder() { }
+        public void AddError(string error, string? propertyName = null) { throw null; }
+        public void AddResult(System.ComponentModel.DataAnnotations.ValidationResult? result) { throw null; }
+        public void AddResults(System.Collections.Generic.IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult?>? results) { throw null; }
+        public void AddResults(ValidateOptionsResult result) { throw null; }
```

Comment:
Is this supposed to be AddResult rather than AddResults?

Reply:
Good question. From the review it was suggested as `AddResults` https://github.com/dotnet/runtime/issues/77404#issuecomment-1438963979 but this may be a typo. @bartonjs can you confirm? I'll change it to be singular as it is more correct.

Reply:
The method can add a plural number of errors but only a singular result. Should probably be `AddResult` to reflect this.